source/app/schedulers/instance_scheduler.py

```diff
@@ -159,11 +171,53 @@ def get_session_for_account(cross_account_role, aws_account):
                     aws_secret_access_key=credentials["SecretAccessKey"],
                     aws_session_token=credentials["SessionToken"],
                 )
-            except Exception as ex:
+            except ClientError as ex:
                 self._logger.error(
-                    ERR_ASSUMING_ROLE.format(cross_account_role, aws_account, str(ex))
+                    "Error Code {}".format(ex.response.get("Error", {}).get("Code"))
                 )
-                return None
+                if ex.response.get("Error", {}).get("Code") == "AccessDenied":
+                    try:
+                        self._logger.error(
+                            "Removing the account {} from scheduling configuration as assume role permission is missing for the iam role {}".format(
+                                aws_account, cross_account_role
+                            )
+                        )
+                        payload = str.encode(
+                            json.dumps(
+                                {
+                                    "account": aws_account,
+                                    "detail-type": "Parameter Store Change",
+                                    "detail": {"operation": "Delete"},
+                                }
+                            )
+                        )
+                        response = self.lambda_client.invoke(
+                            FunctionName=self._context.function_name,
+                            InvocationType="Event",
+                            LogType="None",
+                            Payload=payload,
+                        )
+                        self._logger.info(
+                            "Removing account {} from configuration".format(aws_account)
+                        )
+                        self._logger.debug(
+                            "Lambda response {} for removing account from configuration".format(
+                                response
+                            )
+                        )
```

Comment:
there's a lot of code here in response to this AccessDenied error that I don't understand the purpose of. could this be extracted into a named function that explains what it does (or at the very least have a comment added)?

something like "reportFailure()" or "retryNameOfTask()"

Reply:
if that's what it does, can a comment be added to explain that. My inability to understand this at a glance will likely apply to anybody coming back to read this code in the future.